contracts/AelinPool.sol

```diff
@@ -97,18 +112,64 @@ contract AelinPool is AelinERC20, MinimalProxyFactory, IAelinPool {
             );
-            for (uint256 i = 0; i < _poolData.allowListAddresses.length; i++) {
-                allowList[_poolData.allowListAddresses[i]] = _poolData.allowListAmounts[i];
-                emit AllowlistAddress(_poolData.allowListAddresses[i], _poolData.allowListAmounts[i]);
+            for (uint256 i = 0; i < allowListAddresses.length; i++) {
+                allowList[allowListAddresses[i]] = allowListAmounts[i];
+                emit AllowlistAddress(allowListAddresses[i], allowListAmounts[i]);
             }
             hasAllowList = true;
         }
 
+        NftCollectionRules[] memory nftCollectionRules = _poolData.nftCollectionRules;
+
+        if (nftCollectionRules.length > 0) {
+            // if the first address supports punks or 721, the entire pool only supports 721 or punks
+            if (
+                nftCollectionRules[0].collectionAddress == CRYPTO_PUNKS ||
+                NftCheck.supports721(nftCollectionRules[0].collectionAddress)
+            ) {
+                for (uint256 i = 0; i < nftCollectionRules.length; i++) {
+                    require(
+                        nftCollectionRules[i].collectionAddress == CRYPTO_PUNKS ||
+                            NftCheck.supports721(nftCollectionRules[i].collectionAddress),
+                        "can only contain 721"
+                    );
+                    nftCollectionDetails[nftCollectionRules[i].collectionAddress] = nftCollectionRules[i];
+                    emit PoolWith721(
+                        nftCollectionRules[i].collectionAddress,
+                        nftCollectionRules[i].purchaseAmount,
+                        nftCollectionRules[i].purchaseAmountPerToken
+                    );
+                }
+            }
+            // if the first address supports 1155, the entire pool only supports 1155
+            if (NftCheck.supports1155(nftCollectionRules[0].collectionAddress)) {
+                for (uint256 i = 0; i < nftCollectionRules.length; i++) {
+                    require(NftCheck.supports1155(nftCollectionRules[i].collectionAddress), "can only contain 1155");
+                    nftCollectionDetails[nftCollectionRules[i].collectionAddress] = nftCollectionRules[i];
+
+                    for (uint256 j = 0; j < nftCollectionRules[i].tokenIds.length; j++) {
+                        nftId[nftCollectionRules[i].collectionAddress][nftCollectionRules[i].tokenIds[j]] = true;
+                    }
+                    emit PoolWith1155(
+                        nftCollectionRules[i].collectionAddress,
+                        nftCollectionRules[i].purchaseAmount,
+                        nftCollectionRules[i].purchaseAmountPerToken,
+                        nftCollectionRules[i].tokenIds,
+                        nftCollectionRules[i].minTokensEligible
+                    );
+                }
+            }
+            hasNftList = true;
```

Comment:
as mentioned earlier, I think we need to revert if wrong data is passed in `nftCollectionRules` to avoid creating pools which won't work.

Reply:
@alextheboredape @cranium7811 what should we do here?

Reply:
see fc4f28f90e8709d76f6fe2b19bc85a96efe975ee